src/caffe/net.cpp

```diff
@@ -650,6 +650,61 @@ void Net<Dtype>::ShareTrainedLayersWith(const Net* other) {
 }
 
 template <typename Dtype>
+void Net<Dtype>::ShareBlobsWith(const Net* other) {
+  int num_source_blobs = other->blobs().size();
```

Comment:
OK here's a naive question, since  I'm not sure I understand everything about this change. But the Net::blobs() method just returns the blobs that are outputs of layers. What about the blobs that are the layer's learnt parameters i.e. weights? Do we not want to share those too, so as not to redundantly have to store them as well?
In fact, now that I ask this question, I realize I don't know why there needs to be a test net that's different from the trained net. Why can't we just generate test scores using the current state of the trained net.